app/services/proofing/aamva/request/verification_request.rb

```diff
@@ -0,0 +1,136 @@
+require 'erb'
+require 'faraday'
+require 'rexml/document'
+require 'rexml/xpath'
+require 'securerandom'
+require 'retries'
+
+module Proofing
+  module Aamva
+    module Request
+      class VerificationRequest
+        CONTENT_TYPE = 'application/soap+xml;charset=UTF-8'.freeze
+        DEFAULT_VERIFICATION_URL =
+          'https://verificationservices-cert.aamva.org:18449/dldv/2.1/online'.freeze
+        SOAP_ACTION =
+          '"http://aamva.org/dldv/wsdl/2.1/IDLDVService21/VerifyDriverLicenseData"'.freeze
+
+        extend Forwardable
+
+        attr_reader :config, :body, :headers, :url
+
+        def initialize(config:, applicant:, session_id:, auth_token:)
+          @config = config
+          @applicant = applicant
+          @transaction_id = session_id
+          @auth_token = auth_token
+          @url = verification_url
+          @body = build_request_body
+          @headers = build_request_headers
+        end
+
+        def send
+          with_retries(max_tries: 2, rescue: [Faraday::TimeoutError, Faraday::ConnectionFailed]) do
+            Response::VerificationResponse.new(
+              http_client.post(url, body, headers),
+            )
+          end
+        rescue Faraday::TimeoutError, Faraday::ConnectionFailed => err
+          message = "AAMVA raised #{err.class} waiting for verification response: #{err.message}"
+          raise ::Proofer::TimeoutError, message
+        end
+
+        def verification_url
+          config.verification_url || DEFAULT_VERIFICATION_URL
+        end
+
+        private
+
+        attr_reader :applicant, :transaction_id, :auth_token
+
+        def http_client
+          Faraday.new(request: { open_timeout: timeout, timeout: timeout }) do |faraday|
+            faraday.adapter :net_http
+          end
+        end
+
+        def add_user_provided_data_to_body
+          document = REXML::Document.new(body)
+          user_provided_data_map.each do |xpath, data|
+            REXML::XPath.first(document, xpath).add_text(data)
+          end
+          @body = document.to_s
+        end
+
+        def build_request_body
+          renderer = ERB.new(request_body_template)
+          @body = renderer.result(binding)
+          add_user_provided_data_to_body
+        end
+
+        def build_request_headers
+          {
+            'SOAPAction' => SOAP_ACTION,
+            'Content-Type' => CONTENT_TYPE,
+            'Content-Length' => body.length.to_s,
+          }
+        end
+
+        def document_category_code
+          case applicant.state_id_data.state_id_type
+          when 'drivers_license'
+            '1'
+          when 'drivers_permit'
+            '2'
+          when 'state_id_card'
+            '3'
+          end
+        end
+
+        def message_destination_id
+          return 'P6' if config.cert_enabled.to_s == 'true'
```

Comment:
Hi @solipet, hope you don't mind me asking a question here.  What's the motivation for returning 'P6' here?  Just wondering because I'm working on a PR that touches this area of the codebase.  Thanks! 🙏🏻

Reply:
Hi @eileen-nava,

I don't know - this PR merely moved the code from a separate gem repository into the idp codebase. Looking at [the first PR from that repo](https://github.com/18F/identity-aamva-api-client-gem/pull/1), it appears that P6 was there from the beginning. It appears to be a mock state_id_jurisdiction used for testing (given the update in [this PR](https://github.com/18F/identity-aamva-api-client-gem/pull/7/files))?

Reply:
Hi @solipet, gotcha!  I think you're correct that it is a mock `state_id_jurisdiction` used for testing.  I'll ask in a broader channel to confirm, just in case.